crates/cadenza-eval/STATUS.md

```diff
@@ -219,6 +219,350 @@ let foo =
+- [ ] Proper source tracking for interpolated parts
+
+**Notes**: Common in many modern languages (JavaScript, Kotlin, Swift, etc.). Should convert all expressions to strings automatically.
+
+### Rational Numbers
+
+Support for exact rational number arithmetic to avoid integer truncation and floating-point precision issues.
+
+**Syntax**:
+```cadenza
+let v = 1 / 2  # does not immediately divide and truncate the integer
+let a = v * 2
+assert a == 1 "expected exact rational arithmetic: 1/2 * 2 should equal 1"
+```
+
+**Requirements**:
+- [ ] Implement `Rational` type (likely using a numerator/denominator pair)
+- [ ] Add `Value::Rational` variant
+- [ ] Update division operator to return rationals when dividing integers
+- [ ] Implement rational arithmetic (+, -, *, /)
+- [ ] Implement comparison operators for rationals
+- [ ] Auto-simplification of rationals (reduce to lowest terms)
+- [ ] Conversion between integers, floats, and rationals
+- [ ] Display formatting for rationals
+
+**Notes**: Essential for exact arithmetic. Consider using existing Rust crates like `num-rational`. Should integrate with type system and unit system.
+
+### Quote and Unquote
+
+Support for quoting expressions to create AST values and unquoting to evaluate them.
+
+**Syntax**:
+```cadenza
+let foo = 1
+let ast = quote foo
+unquote ast  # returns 1
+```
+
+**Requirements**:
+- [ ] Implement `quote` macro that prevents evaluation and returns AST
+- [ ] Add `Value::Ast` or similar variant to hold unevaluated expressions
+- [ ] Implement `unquote` macro to evaluate AST values
+- [ ] Proper environment capture and hygiene
+- [ ] Handle nested quote/unquote correctly
+
+**Notes**: Foundation for metaprogramming and macros. Similar to Lisp's quote/unquote or Rust's `quote!` macro.
+
+### Quote with Splicing (Unquote within Quote)
+
+Support for embedding evaluated expressions within quoted code using unquote.
+
+**Syntax**:
+```cadenza
+let foo = 1
+let ast = quote foo
+let ast2 = quote
+    (unquote ast) + (unquote ast)
+unquote ast2  # returns 2
+```
+
+**Requirements**:
+- [ ] Extend quote macro to detect and handle `unquote` within quoted expressions
+- [ ] Implement splicing mechanism to substitute evaluated expressions into AST
+- [ ] Handle nested quote/unquote combinations
+- [ ] Maintain proper source location information through splicing
+- [ ] Test complex splicing scenarios
+
+**Notes**: Enables template-like code generation. Must handle hygiene correctly to avoid variable capture issues.
+
+### User-Defined Macros
+
+Support for user-defined compile-time macros with proper hygiene and environment capture.
+
+**Syntax**:
+```cadenza
+macro add_one expr =
+    # Macros are functions that take unevaluated AST arguments and return an AST
+    let one = quote 1
+    quote
+        (unquote expr) + (unquote one)
+
+add_one 5  # expands to 5 + 1, returns 6
+```
+
+**Requirements**:
+- [ ] Add `macro` keyword to lexer and parser
```

Comment:
we don't need any keywords. the lexer and parser are just aware of identifiers. It's up to the environment to add semantics for those identifiers.